cognitedata/reveal
# feat(react-components): asset mapping click handling and "mapped"-styling

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/UX-1583

## Description :pencil:


Add click-handling support for nodes that are asset-mapped through classic asset mappings.

Note that this code is not optimized for _many_ asset mappings, and assumes filtering them locally is feasible for e.g. on-click operations.

Also, this breaks the API of the `useClickedNode` hook. There may be a way to better future proof this, open to suggestions.

## How has this been tested? :mag:



In storybook, HighlightNode.

## Test instructions :information_source:



Run HighlightNode. Replace the model in the code by 
```
{
        modelId: 1791160622840317,
        revisionId: 502149125550840,
}
```

Click on different nodes (the asset-mapped ones should be highlighted), and see that the console logs the data from the nodes.

## Checklist :ballot_box_with_check:
